### PR TITLE
CLI command for client creation #94

### PR DESCRIPTION
**Pull request by IljaN**
Issue #94

### CODE REVIEW

**Comment by SamuAlfageme** [comment]
@IljaN could we also include the option to "Set `access_token` expiration time" via CLI. Is such command reasonable @DeepDiver1975?

File: lib/Command/CreateClient.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		parent::__construct($name);
+	}
+
+	protected function configure() {
+		$this
+			->setName('oauth2:create-client')
+			->setDescription('Create a OAuth2 client (admin action)')
+			->addArgument('name', InputArgument::REQUIRED, 'Name of the client')
+			->addArgument('redirect-uri', InputArgument::REQUIRED, 'Redirect Url')
+			->addOption('identifier', 'I', InputOption::VALUE_OPTIONAL, "Random generated, set manually for testing only!")
```
Comment (SamuAlfageme) [inline lib/Command/CreateClient.php]: Somehow it's not working as optional for me: https://asciinema.org/a/uiCKTvla2tQQ8NNUUXcZMRFct

```
$ occ oauth2:create-client test http://localhost:*  

  [Doctrine\DBAL\Exception\NotNullConstraintViolationException]
  An exception occurred while executing 'INSERT INTO "oc_oauth2_clients"("secret","name","redirect_uri","allow_subdomains") VALUES(?,?,?,?)' with params ["ooWibiRzcxgPzThxVhBChbCdlitsMIoQrVG2X8wYc20A2HgDxH2Pz3HklQ9fXYBF", "test_branding", "http:\/\/localhost:*", false]
  :
  SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: oc_oauth2_clients.identifier
```

File: lib/Command/CreateClient.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			$output->writeln('<error>A client with this name does already exist.</error>');
+			return;
+		}
+
+		$clientData = [
+			'Name' => $client->getName(),
+			'Identifier' => $client->getIdentifier(),
+			'Secret' => $client->getSecret(),
+			'Redirect Uri' => $client->getRedirectUri(),
+			'Allow Sub-Domains' => $client->getAllowSubdomains()
```
Comment (SamuAlfageme) [inline lib/Command/CreateClient.php]: Could an explicit `false` be stored for this one? The empty field seen fron an admin POV it's a bit weird. 

```
Name: test_branding
Identifier: test_branding
Secret: 0rdlzOb60C6R5uhxPSMEqC9Wx22y3VjklnRT8BdljHfly11GmaP8G1b42rqqxjrF
Redirect Uri: http://localhost:*
Allow Sub-Domains:
```

**Comment by mmattel** [comment]
@IljaN any news in this PR ?

**Comment by michaelstingl** [comment]
still needed with https://github.com/owncloud/oauth2/commit/865f6231055d75ac18c2a84c083b5992ae680f94 ? /cc @DeepDiver1975

**Pull request closed without merging**